trie/ctrie/ctrie.go

```diff
@@ -340,20 +339,80 @@ func (c *Ctrie) ReadOnlySnapshot() *Ctrie {
 	}
 }
 
+// Clear removes all keys from the Ctrie.
+func (c *Ctrie) Clear() {
+	for {
+		root := c.readRoot()
+		gen := &generation{}
+		newRoot := &iNode{
+			main: &mainNode{cNode: &cNode{array: make([]branch, 0), gen: gen}},
+			gen:  gen,
+		}
+		if c.rdcssRoot(root, gcasRead(root, c), newRoot) {
+			return
+		}
+	}
+}
+
+// Iterator returns a channel which yields the Entries of the Ctrie.
+func (c *Ctrie) Iterator() <-chan *Entry {
+	ch := make(chan *Entry)
+	snapshot := c.ReadOnlySnapshot()
+	go func() {
+		traverse(snapshot.root, ch)
+		close(ch)
+	}()
+	return ch
+}
```

Comment:
Damn, you're right, this is problematic because if the user doesn't read from the channel the goroutine will block. Any ideas on making a nicer iterator which doesn't have this problem?